compose/core-compose/src/main/java/com/squareup/workflow/ui/compose/CompositionRoot.kt

```diff
@@ -86,8 +86,8 @@ fun ViewRegistry.withCompositionRoot(root: CompositionRoot): ViewRegistry =
     // If the ambient is false, this is the first time this function has appeared in the composition
     // so far. We provide a true value for the ambient for everything below us, so any recursive
     // calls to this function will hit the if case above and not re-apply the wrapper.
-    Providers(HasViewFactoryRootBeenApplied provides true) {
-      val safeRoot: CompositionRoot = remember(root) { safeCompositionRoot(root) }
+    CompositionLocalProvider(HasViewFactoryRootBeenApplied provides true) {
+      val safeRoot: CompositionRoot = remember(root as Any?) { safeCompositionRoot(root) }
```

Comment:
Why the cast?

Reply:
<img width="1032" alt="Screen Shot 2021-03-13 at 12 58 29 PM" src="https://user-images.githubusercontent.com/1223008/111044178-d7791a00-83fb-11eb-836a-d3fb2140a898.png">

Reply:
Ah yes, this is such a weird bug.